design-documents/graph-ql/coverage/shared-cart.md

```diff
@@ -1,30 +1,45 @@
  ## Problem statement   
- 
-Registered customer should be able to work with the same shopping cart using multiple devices. For example, add products to cart using a laptop and complete checkout using a smartphone.
 
+Once a customer logs in (uses it's token), an existing guest cart needs to be transferred/merged/imported to the customer cart.
+Also the registered customer should be able to work with the same shopping cart on multiple devices that we use that token once we 'login', for example, add products to cart using a laptop and complete checkout using a smartphone using the same customer.
+We used the term 'login' in quotes, because there's really no state/session in graphql.
 
  ## Proposed solution
 
-To achieve desired behavior there should be a way to retrieve active customer cart ID from the server. Based on the assumption that there could be only one active cart per customer at any given moment this can be done as follows:
+To achieve desired behavior we need two operations: 
+
+ 1. **Get customer cart ID:** There should be a way to retrieve active customer cart ID from the server without using the actual cart Id as a parameter when we use a customer. This only works if we have only one active cart per customer at any given moment.
```

Comment:
We should note that, in the case that the customer does not have an active cart, one will be created and returned. And in that case, does it make sense for this to be a mutation, since it could have side effects.